internal/querycoordv2/balance/rowcount_based_balancer.go

```diff
@@ -137,121 +141,82 @@ func (b *RowCountBasedBalancer) balanceReplica(replica *meta.Replica) ([]Segment
 			)
 			stoppingNodesSegments[nid] = segments
 		} else {
-			nodesSegments[nid] = segments
+			onlineNodesSegments[nid] = segments
 		}
 
-		cnt := 0
 		for _, s := range segments {
-			cnt += int(s.GetNumOfRows())
+			totalCnt += int(s.GetNumOfRows())
 		}
-		nodesRowCnt[nid] = cnt
-		totalCnt += cnt
 	}
 
 	if len(nodes) == len(stoppingNodesSegments) {
 		// no available nodes to balance
 		return nil, nil
 	}
 
-	if len(nodesSegments) == 0 {
-		return nil, nil
-	}
-
-	average := totalCnt / len(nodesSegments)
-	neededRowCnt := 0
-	for nodeID := range nodesSegments {
-		rowcnt := nodesRowCnt[nodeID]
-		if rowcnt < average {
-			neededRowCnt += average - rowcnt
-		}
-	}
-
-	if neededRowCnt == 0 {
+	if len(onlineNodesSegments) == 0 {
```

Comment:
Is this expected? this will lead stopping to balance the segments in stopping nodes

Reply:
if all nodes down, no need to do any balance here. but this seems to be a duplicate check with above lines.